### DISCUSSION TITLE
FAQ - Why is youtube video not working in Hubs?

### DISCUSSION THREAD

**Discussion by emclaren** [mozilla/hubs#2681] [General Questions & Discussion] [answered]
My youtube video is showing up as a screenshot in Hubs. It used to play without issue.

**Comment by emclaren** [comment, marked as answer]
YouTube regularly blocks Hubs servers due to the number of incoming requests. We recommend not relying on youtube for critical content at your event. You could try saving your video on another platform like Vimeo, or saving it to Dropbox or to a personal server and streaming it from there.

**Reply by BonzoMartini** [reply]
Sometimes, Spoke and Hubs will link to vimeo videos, sometimes it will not. With Vimeo videos Im getting an error on some videos however the 'error loading project' window wont allow me to copy text out but the gist of it is "error loading video..... failed to load init decoder." **Any clues on how to fix??**

**Reply by ocularvr** [reply]
Would it not be optimal to upload the videos on your hubs server then cloudflare it? surely this will be sufficient - more expensive but mission critical

**Reply by davegoopot** [reply]
@ocularvr , do you mean upload the whole video file into the Hub? How do you do that? Is that supported on hubs.mozilla.com, or only for self-hosted versions?

**Reply by brunomartelli** [reply]
you can add video as a file, however my vids are large, so i assume ppl would prefer to stream

**Reply by ocularvr** [reply]
#davegoopot, yeah thats what I meant. With my implementation, YouTube works on and off so if its an important video its better to upload it on the hubs server. 

The way to do it

Step 1 - Login to your hubs
Step 2 - Go to Spoke
Step 3 - In the tab on the bottom left, look for "My assets"
Step 4 - Click the blue button (i think same color for all) that reads "Upload..."
Step 5 - Upload your desired video
Step 6 - Select that video and place it in a scene 

This is one way of doing it. If you want to render the video in the middle of a talk, then just do this:

Look at the Properties tab on the right of the video you just uploaded, look for "Video Url" copy this. When in your room, select the magic wand (bottom right of that box), then add the url you copied which is your video on your server. You and just pop it into view like that also. This has nothing to do with your question but could be useful.

**Reply by davegoopot** [reply]
I believe the size limit to video upload is 128MB which is obviously limiting for video.
See: https://hubs.mozilla.com/docs/spoke-adding-scene-content.html

**Comment by davegoopot** [comment]
I've experimented with some of these and the results are patchy.

Nivrana unplugged, https://www.youtube.com/watch?v=z9LiPuVRyU8 , doesn't play
BBC Elbow Glastonbury, https://www.youtube.com/watch?v=sQQFjfbWF6M , plays
U2 With or without you, https://youtu.be/ujNeHIo7oTE , doesn't play
Dr Buttler (local band), https://youtu.be/6gMl0kbsy5U , plays
TED talk, https://www.youtube.com/watch?v=5MgBikgcWnY , plays

Something to do with the licence maybe? I don't know how to see the licence information from the YouTube web page.

**Comment by emclaren** [comment]
Following up with a bit of additional information related to live streaming as we gets asked about it a lot. 

You can use platforms like Twitch & Youtube for live streaming, however, depending on the number of folks attempting to stream into Hubs concurrently, we don't guarantee that third party software like Twitch will work (if too many concurrent requests are made, servers are sometimes blocked, this happens quite frequently with youtube). 

You should be able to use any streaming service with an HLS endpoint, we've heard of people using paid services for doing live streams into Hubs (for example: Wowza). This type of service doesn't have the same type of capacity limits for our servers to make requests for video feeds. This also has an added benefit of preventing people from watching the performances on Twitch instead of in your Hubs space. 

I would recommend doing some testing in advance of the event to ensure that live streams load properly. I have no reason to suspect that they wouldn't, however, I haven't worked with live streaming platforms myself so i'm not sure how difficult they are to set up.
